src/DotLiquid/StandardFilters.cs

```diff
@@ -640,16 +640,27 @@ public static string Date(Context context, object input, string format)
                 return input.ToString();
 
             DateTimeOffset dateTimeOffset;
+            long timestamp = 0L;
             if (input is DateTimeOffset inputOffset)
             {
                 dateTimeOffset = inputOffset;
             }
-            else if ((input is decimal) || (input is double) || (input is float) || (input is int) || (input is uint) || (input is long) || (input is ulong) || (input is short) || (input is ushort))
+            else if ((input is decimal) || (input is double) || (input is float) || (input is int) || (input is uint) || (input is long) || (input is ulong) || (input is short) || (input is ushort)
+                || (input is string stringInput && long.TryParse(stringInput, NumberStyles.Integer, CultureInfo.InvariantCulture, out timestamp)))

```

Comment:
```suggestion
                || (input is string stringInput && long.TryParse(stringInput, NumberStyles.Integer, CultureInfo.InvariantCulture, out var timestamp)))

```

Reply:
I tried this originally, but it doesn't compile: `The name 'timestamp' does not exist in the current context [DotLiquid]csharp(CS0103)`
It seems the compiler is not aware that `out var timestamp` will be initialised when `input is string`.

Reply:
I see, you use timestamp when input is not string by assignment.